pkg/cloudprovider/providers/azure/azure.go

```diff
@@ -0,0 +1,143 @@
+package azure
+
+import (
+	"encoding/json"
+	"io"
+
+	"k8s.io/kubernetes/pkg/cloudprovider"
+
+	"github.com/Azure/azure-sdk-for-go/arm/compute"
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/golang/glog"
+)
+
+const AzureProviderName = "azure"
+
+type AzureConfig struct {
+	Cloud             string `json:"cloud"`
```

Comment:
And presumably change the json decode in `azure.go` to yaml so we can allow both; sounds good.